src/app/App.js

```diff
@@ -26,37 +25,32 @@ class App extends Component {
     )
   }
 
-  displayMovieDetails = (id) => {
-    this.setState({
-      displayMovieDetails: this.state.movies.find(movie => movie.id === id)
-    })
-  }
-
   displayMovieLibrary = () => {
     this.setState({
       displayMovieDetails: null
     })
   }
 
-  checkForErrors = response => {
-    if (!response.ok) {
-      this.setState( {error: 'Oop, that/s no good 😅'} )
-    } else {
-      return response.json();
-    }
+  displayMovieDetails = (id) => {
+    fetchMovieDetails(id)
+      .then(movieDetails => this.setState({displayMovieDetails: movieDetails.movie}))
+      .catch(err => this.setState({ error: 'There was a problem loading this title. Try again later.'}))
   }
 
   componentDidMount() {
-    fetch('https://rancid-tomatillos.herokuapp.com/api/v2/movies')
-      .then(this.checkForErrors)
-      .then(data => {
-        this.setState( {
-          movies: data.movies
-        })
-      })
-      .catch(err => this.setState({ error: 'Oh wow. This is embarassing. Try again later? 😅' }))
+    fetchAllMovies()
+      .then(movieData => this.setState({movies: movieData.movies}))
+      .catch(err => this.setState({ error: 'Oh wow. This is embarassing. Try again later? 😅'}))
   }
 
+  // checkForErrors = response => {
+  //   if (!response.ok) {
+  //     this.setState( {error: 'Oop, that/s no good 😅'} )
+  //   } else {
+  //     return response.json();
+  //   }
+  // }
+
 }
```

Comment:
This is the syntax for checking server-side errors that would occur if the server itself was down. I just broke it out into a separate function so the network request wouldn't get too long. We should move it over into the new network request files and ask Scott and Leta about it.